pkg/engine/utils.go

```diff
@@ -38,32 +40,31 @@ type EngineStats struct {
 	RulesAppliedCount int
 }
 
-func checkKind(kinds []string, resourceKind string, gvk schema.GroupVersionKind) bool {
+func checkKind(subresourceGVKToAPIResource map[string]*metav1.APIResource, kinds []string, gvk schema.GroupVersionKind, subresourceInAdmnReview string) bool {
 	title := cases.Title(language.Und, cases.NoLower)
+	var result = false
 	for _, k := range kinds {
-		parts := strings.Split(k, "/")
-		if len(parts) == 1 {
-			if k == "*" || resourceKind == title.String(k) {
-				return true
-			}
-		}
-
-		if len(parts) == 2 {
-			kindParts := strings.SplitN(parts[1], ".", 2)
-			if gvk.Kind == title.String(kindParts[0]) && gvk.Version == parts[0] {
-				return true
+		if k != "*" {
+			gv, kind := kubeutils.GetKindFromGVK(k)
+			apiResource, ok := subresourceGVKToAPIResource[k]
+			if ok {
+				result =
+					apiResource.Group == gvk.Group && (apiResource.Version == gvk.Version || strings.Contains(gv, "*")) && apiResource.Kind == gvk.Kind
+			} else { // if the kind is not found in the subresourceGVKToAPIResource, then it is either not a subresource or the subresource is not registered in the cluster
+				result = title.String(kind) == gvk.Kind && subresourceInAdmnReview == ""
```

Comment:
What does this mean?

Reply:
I removed then end part, that condition won't be reached but this simply means:
```go
apiResource, ok := subresourceGVKToAPIResource[k]
if ok {
	result =
		apiResource.Group == gvk.Group && (apiResource.Version == gvk.Version || strings.Contains(gv, "*")) && apiResource.Kind == gvk.Kind
} else { // if the kind is not found in the subresourceGVKToAPIResource, then it is not a subresource
	result = title.String(kind) == gvk.Kind && subresourceInAdmnReview == ""
	if gv != "" {
		result = result && kubeutils.GroupVersionMatches(gv, gvk.GroupVersion().String())
	}
}
```
if `apiResource` is not present in the subresource map then treat it as a normal resource.